lib/push-jenkins-update.js

```diff
@@ -0,0 +1,80 @@
+'use strict'
+
+const githubClient = require('./github-client')
+
+const jenkinsToGhStatusMap = {
+  'SUCCESS': {
+    state: 'success',
+    message: 'all tests passed'
+  },
+  'FAILURE': {
+    state: 'failure',
+    message: 'build failure'
+  },
+  'ABORTED': {
+    state: 'error',
+    message: 'build aborted'
+  },
+  'UNSTABLE': {
+    state: 'error',
+    message: 'build unstable'
+  }
+}
+
+const rxDisplayName = /node\-test\-commit\-(\w+)/
+
+function pushJenkinsUpdate (options, build) {
+  const statusOpts = extendWithBuildData(options, build)
+  const createGhStatus = createGhStatusFn(statusOpts)
+  const isPending = build.result === null
+  const matchedGhStatus = jenkinsToGhStatusMap[build.result]
+
+  if (isPending) {
+    createGhStatus('pending', 'build in progress')
```

Comment:
Sure, we could trust whatever curl pushes, but keep in mind the state (first param) has to be one of `pending|success|error|failure`.

Some explanation of what's provided when invoking `githubClient.statuses.create(..)` to set the github status and what's displayed on github.com:

- **state**: what decides the icon/color to the left
- **context**: text in the middle in bold
- **description**: plain text to the right

![image](https://cloud.githubusercontent.com/assets/1231635/15094604/d1168e2a-14a9-11e6-973b-e3332ef56760.png)